protocol/nats_jetstream/v3/protocol.go

```diff
@@ -0,0 +1,131 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	"github.com/nats-io/nats.go"
+)
+
+// Protocol is a reference implementation for using the CloudEvents binding
+// integration. Protocol acts as both a NATS client and a NATS handler.
+type Protocol struct {
+
+	// nats connection
+	conn      *nats.Conn
+	url       string
+	natsOpts  []nats.Option
+	connOwned bool
+
+	receiver *receiver
+
+	sender *sender
+}
+
+// NewProtocol creates a new NATS protocol.
```

Comment:
Elaborate more on the protocol constructor please. Also, don't use newlines, just keep describing.


```suggestion
// NewProtocol creates a new protocol using NATS JetStream. [... <describe the default behavior and which options to use/which conflict>]
```